src/_pytest/compat.py

```diff
@@ -20,6 +20,16 @@
 import attr
 import py
 
+# fmt: off
+# Workaround for https://github.com/sphinx-doc/sphinx/issues/10351.
+# If `overload` is imported from `compat` instead of from `typing`,
+# Sphinx doesn't recognize it as `overload` and the API docs for
+# overloaded functions look good again. But type checkers handle
```

Comment:
IIUC, should this read:

```suggestion
# Sphinx doesn't recognize it as `overload` and the API docs for
# overloaded functions look bad. But type checkers handle
```

?

Reply:
The problem is that Sphinx started recognizing overloads and rendering them, but the way it does it has some issues (see picture in https://github.com/sphinx-doc/sphinx/issues/10351, pretty similar to the we cleaned up a while back). So this hack actually makes things look good by making Sphinx *not* recognize `overload` :)

Reply:
AHH I see, thanks!